probes/dns/dns_test.go

```diff
@@ -100,7 +104,50 @@ func TestRun(t *testing.T) {
 	if err := p.Init("dns_test", opts); err != nil {
 		t.Fatalf("Error creating probe: %v", err)
 	}
-	runProbe(t, "basic", p, 1, 1)
+	runProbe(t, "basic", p, nil, 1, 1)
+}
+
+func TestResolveFirst(t *testing.T) {
+	p := &Probe{}
+	resolveF := func(target string, ipVer int) (net.IP, error) {
+		if target == "foo" {
+			return net.ParseIP("8.8.8.8"), nil
+		}
+		return nil, fmt.Errorf("unexpected target: %v", target)
+	}
+	resolveFirst := true
+	opts := &options.Options{
+		Targets:  targets.StaticTargets("foo"),
+		Interval: 2 * time.Second,
+		Timeout:  time.Second,
+		ProbeConf: &configpb.ProbeConf{
+			ResolveFirst: &resolveFirst,
+		},
+	}
+	if err := p.Init("dns_test_resolve_first", opts); err != nil {
+		t.Fatalf("Error creating probe: %v", err)
+	}
+	runProbe(t, "basic", p, resolveF, 1, 1)
+}
+
+func TestResolveFirstError(t *testing.T) {
```

Comment:
These two tests can likely be combined into one function to avoid some duplication:
```
func TestResolveFirst(t *testing.T) {
  p := ..
  ...
 
 t.Run("dns_test_resolve_first", func (t *testing.T) {
   resolveF := func(..) () { ... }
   runProbe(....)
})

t.Run("dns_test_resolve_first_error", func (t *testing.T) {
   resolveF := func(..) () { ... }
   runProbe(....)
})
```